### PR TITLE
Don't open dropdown on ESC on trigger element

### LINKED ISSUE #28751: Dropdowns should not open/trigger on "ESC"
This seems to be a regression from 3.x (see https://github.com/twbs/bootstrap/issues/14646)

Currently, dropdowns https://getbootstrap.com/docs/4.3/components/dropdowns/ can be opened by setting focus to the trigger element and pressing the "ESC" key. It's not a major problem, but non-standard and unexpected behaviour, which should really be removed.

### PR DESCRIPTION
**Pull request by patrickhlauke**
Closes #28751

(doesn't need porting to v5, as there the issue doesn't appear to be present anymore)

### CODE REVIEW

**Comment by Johann-S** [comment]
How is it possible it's not present in v5 ? 😲

File: js/src/dropdown.js [hunk 1]
```diff
@@ -475,6 +475,10 @@ class Dropdown {
     const parent   = Dropdown._getParentFromElement(this)
     const isActive = $(parent).hasClass(ClassName.SHOW)
 
+    if (!isActive && event.which === ESCAPE_KEYCODE) {
```
Comment (Johann-S) [inline js/src/dropdown.js]: Can you add the unit test for this check ?

**Comment by patrickhlauke** [comment]
> How is it possible it's not present in v5 ? 😲

no idea. all i know is that when i tried running current v5, it didn't exhibit the behavior (unless i messed up somewhere along the line)

**Comment by Johann-S** [comment]
I'm very surprise the v5 code follow the v4 code, it shouldn't have such difference, is it possible for you to list what changed between v4 and v5 ?

**Comment by patrickhlauke** [comment]
> I'm very surprise the v5 code follow the v4 code, it shouldn't have such difference, is it possible for you to list what changed between v4 and v5 ?

incidentally, i just added the new unit test to the clean master here locally, and ran the tests ... and they all passed. will dig deeper into why v5 is unaffected...

**Review by Johann-S — APPROVED** [review]

**Comment by Johann-S** [comment]
Thanks @patrickhlauke 👍 Let me know if you have any questions about the v5 code

**Comment by patrickhlauke** [comment]
> Thanks @patrickhlauke 👍 Let me know if you have any questions about the v5 code

without digging too deeply, one change that seems relevant is that in dropdown.js in v4 on line 484 it triggers a "click" (which happens on keydown for ESC as well) whereas in v5 on the matching line 492 is does `Dropdown._clearMenus()` which, from what I can see, does all sorts of extra sanity checking.

**Comment by patrickhlauke** [comment]
and for reference, the new issue i found in v5 https://github.com/twbs/bootstrap/issues/28914

**Comment by Johann-S** [comment]
Hmm true, nice catch, I don't remember why we made that change 🤔

**Comment by patrickhlauke** [comment]
> Hmm true, nice catch, I don't remember why we made that change 🤔

well, a real button already reacts to all the appropriate things like SPACE and ENTER to open the dropdown (by firing the click event), so unnecessary for that. this leaves keydown reaction when menu is already open, which can really only be cursor keys and esc (plus special cases where somebody stuck an input/textarea in the menu itself) ... so i guess this makes more sense.

**Pull request merged**